src/app/grid/grid-select/grid-select-sample.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Character } from './characters';
+import { Component, OnInit, ViewChild } from "@angular/core";
+import { IgxDialogComponent, IgxGridComponent, IGridCellEventArgs } from "igniteui-angular";
+import { DATA, ALIGNMENTS, RACES, CLASSES } from "./data";
+
+@Component({
+    selector: "app-grid-select-sample",
+    styleUrls: ["./grid-select-sample.component.scss"],
+    templateUrl: "./grid-select-sample.component.html"
+})
+export class GridSelectComponent implements OnInit {
+
+    @ViewChild("grid1", { read: IgxGridComponent, static: true })
+    public grid1: IgxGridComponent;
+    @ViewChild("dialogAdd", { read: IgxDialogComponent, static: true })
+    public dialog: IgxDialogComponent;
+    public data;
+    public alignments;
+    public races;
+    public classes;
+    public character;
+
+    public ngOnInit() {
+        this.data = DATA.map((e) => {
+            const indexAlignments = Math.floor(Math.random() * ALIGNMENTS.length);
+            e.alignment = ALIGNMENTS[indexAlignments];
```

Comment:
Setting random data seems nice, though it is not strictly needed for the purposes of the sample. If you would like to keep it, I suggest extracting this in a separate method and call this method in the ngOninit handler. Something like:
    public ngOnInit() {
        this.data = this.getRandomData()....
        
This will make the sample more clean&readable from a user perspective.

Reply:
Moved the logic to a new function as suggested.

Reply:
Could cleanup the empty line between:

    public generateRandomData(data) {

        return data.map((e) => {